src/test/java/org/codehaus/plexus/archiver/zip/ArchiveFileComparator.java

```diff
@@ -1,202 +1,169 @@
-            TarArchiveEntry ze = (TarArchiveEntry) en.nextElement();
-            if ( !ze.isDirectory() )
-            {
-                consumer.accept( ze );
-            }
-        }
+        Collections.list( file.getEntries() )
+            .stream()
+            .filter( entry -> !entry.isDirectory() )
+            .forEach( consumer );
     }
 
     /**
-     * Creates a map with the archive files contents. The map keys
-     * are the names of file entries in the archive file. The map
-     * values are the respective archive entries.
+     * Creates a map with the archive files contents. The map keys are the names of file entries in the archive file.
+     * The map values are the respective archive entries.
      */
-    private static Map<String, TarArchiveEntry> getFileEntries( ArchiveFile file )
-        throws IOException
+    private static Map<String, TarArchiveEntry> getFileEntries( TarFile file )
     {
-        final Map map = new HashMap();
-        for ( java.util.Enumeration en = file.getEntries(); en.hasMoreElements(); )
+        try
         {
-            TarArchiveEntry ze = (TarArchiveEntry) en.nextElement();
-            if ( ze.isDirectory() )
-            {
-                continue;
-            }
-            if ( map.put( ze.getName(), ze ) != null )
-            {
-                Assert.fail( "Multiple archive file entries named " + ze.getName() + " found." );
-            }
+            return Collections.list( file.getEntries() )
+                .stream()
+                .filter( entry -> !entry.isDirectory() )
+                .map( TarArchiveEntry.class::cast )
+                .collect( Collectors.toMap( ArchiveEntry::getName, Function.identity() ) );
+        }
+        catch ( IOException e )
+        {
+            throw new UncheckedIOException( e );
         }
-        return map;
     }
 
-    private static Map getFileEntries( ZipFile file )
-        throws IOException
+    private static void assertTarEquals( TarFile file1, TarArchiveEntry entry1,
+                                         TarFile file2, TarArchiveEntry entry2 )
     {
-        final Map map = new HashMap();
-        for ( java.util.Enumeration en = file.getEntries(); en.hasMoreElements(); )
+        assertThat( entry1.isDirectory() ).isEqualTo( entry2.isDirectory() ).isFalse();
+        assertThat( entry1.isSymbolicLink() ).isEqualTo( entry2.isSymbolicLink() );
+        assertThat( entry1.getLastModifiedDate() ).isEqualTo( entry2.getLastModifiedDate() );
+
+        try ( InputStream is1 = file1.getInputStream( entry1 );
+              InputStream is2 = file2.getInputStream( entry2 ); )
         {
-            ZipArchiveEntry ze = (ZipArchiveEntry) en.nextElement();
-            if ( ze.isDirectory() )
-            {
-                continue;
-            }
-            if ( map.put( ze.getName(), ze ) != null )
-            {
-                Assert.fail( "Multiple archive file entries named " + ze.getName() + " found." );
```

Comment:
Maybe I'm missing something, but to me it looks like the new version would not verify if that the same entry exists only once in the archive.

Reply:
There is an implicit check with the Collector.toMap as it uses a uniqKeysMapAccumulator that throws an exception on a duplicate key.